src/java/com/puppetlabs/puppetserver/ExecutionStubImpl.java

```diff
@@ -10,25 +14,70 @@
     private static final Logger log = LoggerFactory.getLogger(ExecutionStubImpl.class);
 
     /**
-     * Executes the given command in a separate process.
+     * Takes a prepared CommandLine instance and executes it using some sane
+     * defaults and a DefaultExecutor. Also makes a delicious pancake.
      *
-     * @param command the command to execute
-     * @return A 2-element array; the first element is the output of the process.
-     *      The second element is the exit code of the process.
+     * @param commandLine commandLine instance to execute
+     * @return An ExecutionStubResult with output, error, and the exit code of the process
      *
      * @throws InterruptedException
      * @throws IOException
      */
-    public static ExecutionStubResult executeCommand(String command)
+    private static ExecutionStubResult executeExecutor(org.apache.commons.exec.CommandLine commandLine)
             throws InterruptedException, IOException {
+        DefaultExecutor executor = new DefaultExecutor();
+        ByteArrayOutputStream errStream = new ByteArrayOutputStream();
+        ByteArrayOutputStream outStream = new ByteArrayOutputStream();
+        PumpStreamHandler streamHandler = new PumpStreamHandler(outStream, errStream);
+
+        // Don't throw exception on non-zero exit code
+        executor.setExitValues(null);
 
-        ProcessWrapper wrapper = new ProcessWrapper(Runtime.getRuntime().exec(command));
+        // Set up the handlers
+        executor.setStreamHandler(streamHandler);
+
+        Integer exitCode = executor.execute(commandLine);
+
+        String stdErr = errStream.toString();
```

Comment:
Do we need to close the streams when we're done with them?

Can't remember, been a while since I worked with java streams